src/Sarif.Multitool/Rules/RuleResources.resx

```diff
@@ -192,4 +192,10 @@
   <data name="SARIF1017_InvalidIndex" xml:space="preserve">
     <value>If an object contains a property that is used as an array index, then that array must be present and must contain at least "index + 1" elements.</value>
   </data>
+  <data name="SARIF1018_Default" xml:space="preserve">
```

Comment:
>SARIF1018_Default [](start = 14, length = 17)

SARIF1018 seems like a subset of SARIF1015 or SARIF1014 to me. i.e. I'm not clear why a new rule is required. From a naming perspective it's confusing because SARIF1015 is verifying absolute URIs and even has a check for "Certain URIs are required to be absolute." and SARIF1014 is really the corollary check to SARIF1018.
But if we require a new rule (I'm not familiar with the rule definition strategy for SARIF validation,) the _Default string should probably follow the pattern of the other rules. I was expecting something like (I have no idea what the placeholders are for, I'm just copying the other text for the other rules in the resx):
{0}: This artifactLocation object does not contain a "uriBaseId" property, which means that the value of the "uri" property must be absent or an absolute URI.

Reply:
Certainly reasonable people can disagree on the rule partitioning strategy. I think there are three distinct rules:

1. URIs that must always be absolute, no matter what. (SARIF1015).
2. URIs that occur in `artifactLocation` objects where `uriBaseId` is present: there must always be relative. (New rule, not yet written)
3. `artifactLocation` objects where `uriBaseId` is absent _and that occur in run.originalUriBaseIds_: This is the special case covered by SARIF1018. In this case only, `uri` can either be an absolute URI, or it can be `null`.

I think the partitioning I have is reasonable.

---
In reply to: [314373814](https://github.com/microsoft/sarif-sdk/pull/1644#discussion_r314373814) [](ancestors = 314373814)